debezium-server/debezium-server-infinispan/src/main/java/io/debezium/server/infinispan/InfinispanSinkConsumer.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.infinispan;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.impl.ConfigurationProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.server.BaseChangeConsumer;
+import io.debezium.server.CustomConsumerBuilder;
+
+/**
+ * An implementation of the {@link DebeziumEngine.ChangeConsumer} interface that publishes change event messages to predefined Infinispan cache.
+ *
+ * @author vjuranek
+ */
+@Named("infinispan")
+@Dependent
+public class InfinispanSinkConsumer extends BaseChangeConsumer implements DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InfinispanSinkConsumer.class);
+
+    private static final String CONF_PREFIX = "debezium.sink.infinispan.";
+    private static final String SERVER_HOST = CONF_PREFIX + "server.host";
+    private static final String SERVER_PORT = CONF_PREFIX + "server.port";
+    private static final String CACHE_NAME = CONF_PREFIX + "cache";
+    private static final String USER_NAME = CONF_PREFIX + "user";
+    private static final String PASSWORD = CONF_PREFIX + "password";
+
+    private RemoteCacheManager remoteCacheManager;
+    private RemoteCache cache;
+
+    @Inject
+    @CustomConsumerBuilder
+    Instance<RemoteCache> customCache;
```

Comment:
Could you please check other sinks? I am a bit unsure if this should provide `RemoteCache` or `RemoteCacheManager` is more logical.

Reply:
This would depend if we want to support multiple caches or not. In case of multiple caches `RemoteCacheManager` should be use, in case of a single cache (current state), it's IMHO better to use `RemoteCache`. `RemoteCacheManager` could be used also in this case, but would make the configuration more complicated as the user would have to provide also cache name.